src/pymor/models/basic.py

```diff
@@ -87,6 +87,46 @@ def __str__(self):
     def _compute_solution(self, mu=None, **kwargs):
         return self.operator.apply_inverse(self.rhs.as_range_array(mu), mu=mu)
 
+    @property
+    def dual(self):
+        try:
+            return self._dual
+        except AttributeError:
+            assert self.output_functional is not None
+            assert self.output_functional.linear
+            # TODO: assert that the operator is symmetric
+            self._dual = self.with_(rhs=self.output_functional.H)
+            return self._dual
+
+    def _compute_solution_d_mu(self, parameter, index, solution, mu):
+        residual_dmu_lhs = VectorOperator(self.operator.d_mu(parameter, index).apply(solution, mu=mu))
```

Comment:
Why 'residual'?

Reply:
I agree that it is a misleading name. It originally came from the fact that it is the left hand side of the residual.